Plugins/Mute/Plugin.cs

```diff
@@ -7,97 +7,203 @@ namespace Mute;
 
 public class Plugin : IPlugin
 {
+    public string Name => "Mute";
+    public float Version => (float) Utilities.GetVersionAsDouble();
+    public string Author => "Amos";
+
+    public const string MuteKey = "IW4MMute";
+    public const string MuteListKey = "IW4MMuteList";
+    public static MuteManager MuteManager { get; private set; } = null!;
+    public static readonly Server.Game[] SupportedGames = {Server.Game.IW4};
     private readonly IInteractionRegistration _interactionRegistration;
     private static readonly string MuteInteraction = nameof(MuteInteraction);
 
-    public Plugin(IMetaServiceV2 metaService, IInteractionRegistration interactionRegistration)
+    public Plugin(IMetaServiceV2 metaService, IInteractionRegistration interactionRegistration,
+        ITranslationLookup translationLookup)
     {
         _interactionRegistration = interactionRegistration;
-        DataManager = new DataManager(metaService);
+        MuteManager = new MuteManager(metaService, translationLookup);
     }
 
-    public string Name => "Mute";
-    public float Version => (float) Utilities.GetVersionAsDouble();
-    public string Author => "Amos";
-
-    public static string MuteKey = "IW4MMute";
-
-    public static DataManager DataManager;
-    public static readonly Server.Game[] SupportedGames = {Server.Game.IW4};
-
     public async Task OnEventAsync(GameEvent gameEvent, Server server)
     {
         if (!SupportedGames.Contains(server.GameName)) return;
 
         switch (gameEvent.Type)
         {
             case GameEvent.EventType.Join:
-                switch (await DataManager.ReadPersistentData(gameEvent.Origin))
+                // Check if user has any meta set, else ignore (unmuted)
+                var muteMetaJoin = await MuteManager.GetCurrentMuteState(gameEvent.Origin);
+                switch (muteMetaJoin.MuteState)
                 {
                     case MuteState.Muted:
+                        // Let the client know when their mute expires.
+                        gameEvent.Origin.Tell(Utilities.CurrentLocalization
+                            .LocalizationIndex["PLUGINS_MUTE_REMAINING_TIME"].FormatExt(
+                                muteMetaJoin.Expiration is not null
+                                    ? muteMetaJoin.Expiration.Value.HumanizeForCurrentCulture()
+                                    : Utilities.CurrentLocalization.LocalizationIndex["PLUGINS_MUTE_NEVER"],
+                                muteMetaJoin.Reason));
+
+                        // Handle unmute if expired.
+                        if (await IsExpiredMute(gameEvent.Origin, server)) break;
+
+                        // Handle mute of active mutes.
                         await server.ExecuteCommandAsync($"muteClient {gameEvent.Origin.ClientNumber}");
+                        muteMetaJoin.CommandExecuted = true;
```

Comment:
I don't like that you've not got two places to maintain the mute command execution. I think this should be in the mute manager.